wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/FileUtil.java

```diff
@@ -96,6 +95,21 @@ public static boolean validateFileType(final FileItemWrap newFile, final List<St
 		return extMatch || mimeMatch;
 	}
 
+	/**
+	 * Identify the mime type of a file.
+	 * 
+	 * @param file the File to detect.
+	 * @return mime type as detected by Apache tika, otherwise null.
+	 * @throws IOException if the mime type detection fails.
+	 */
+	public static String getFileMimeType(final File file) throws IOException {
+		if (file != null) {
```

Comment:
This is OK but given this is a public method don't you think it would be safer to throw an illegal argument exception if we don't have a file? After all, the MIME type of `null` is not `null`.

Reply:
Because it is dealing with File, I felt leaving `IOException` is appropriate. Both illegal argument exception and `IOException` are runtime exceptions, so the client need not catch.
But I accept to change to illegal argument exception if it makes more sense.